### DISCUSSION TITLE
Adding @relation more than twice

### DISCUSSION THREAD

**Discussion by amin-wan-kenobi** [prisma/prisma#2191] [Q&A] [answered]
Below is my datamodel.graphql

```
type User {
  id: ID! @id
  email: String! @unique
  gifts: [Gift!]!
}
 
type Gift {
  id: ID! @id
  name: String!
  broughtBy: User!
  givenTo: User!
}
```
Each user can have list of gifts and each gift is made by a user and given to another user. When I try to deploy this it give me below error

> Gift
>     ✖ The relation field `broughtBy` must specify a `@relation` directive: `@relation(name: "MyRelation")`
>     ✖ The relation field `givenTo` must specify a `@relation` directive: `@relation(name: "MyRelation")`

So I want to query my users and be able to see the list of the user gifts and see each gift is brought by who and also query the list of gifts to see each gift is brought by who and given to who.

Maybe from database point of view it would be like this
I am thinking we should have id of the gift, id of the user who received the gift and id of the user who brought the gift.

How can we do this? I tried to do the relation between 3 of them but I get A relation directive cannot appear more than twice.  error.

**Comment by janpio** [comment, marked as answer]
This sounds like a Prisma 1 question. Prisma 1's repository was moved to https://github.com/prisma/prisma1, so you might want to ask your question there again to get an answer.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
